PyEMD/significancetest.py

```diff
@@ -0,0 +1,127 @@
+            counter+=1
+    return counter
+
+#helper function: Find mean period of an IMF
+def mean_period(data):
+    s = sign_change(data)
+    if s != 0:
+        return len(data)/s
+    else:
+        return len(data)
+
+#helper function: find energy of signal/IMF
+def energy(data):
+    return sum(pow(data, 2))
+
+#helper function: find IMF significance in 'a priori' test
+def significance_apriori(energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    if (energy_density <= (-math.log(T) + k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))) and (energy_density >= (-math.log(T) - k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))):
+        return False
+    else:
+        return True
+
+#helper function: find significance in 'a posteriori' test
+def significance_aposteriori(scaled_energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    if (scaled_energy_density <= (-math.log(T) + k*(math.sqrt(2/N)*math.exp(math.log(T)/2)))):
+        return False
+    else:
+        return True
+
+def whitenoisecheck(data: np.ndarray, method: str='EMD',  test: str='aposteriori', rescaling_imf: int=1, alpha: float= 0.95):
+    '''
+    References
+    --------------
+    -- Zhaohua Wu, and Norden E. Huang. “A Study of the Characteristics of White Noise Using the Empirical Mode Decomposition Method.” Proceedings: Mathematical, Physical and Engineering Sciences, vol. 460, no. 2046, The Royal Society, 2004, pp. 1597–611, http://www.jstor.org/stable/4143111.
+
+    Examples
+    --------------
+    >>> import numpy as np
+    >>> from PyEMD.significancetest import whitenoisecheck
+    >>> T = np.linspace(0, 1, 100)
+    >>> S = np.sin(2*2*np.pi*T)
+    >>> significant_imfs = whitenoisecheck(S, test='apriori')
+    >>> significant_imfs
+    {1: 0, 2: 1}
+    >>> type(significant_imfs)
+    <class 'dict'>
+    '''
+    assert alpha > 0 and alpha < 1, "alpha value should be in between (0,1)"
+    assert method == 'EMD' or method == 'EEMD' or method == 'CEEMDAN', 'Invalid decomposition method'
+    assert test == 'apriori' or test == 'aposteriori', 'Invalid test type'
+    assert isinstance(data, np.ndarray), 'Invalid Data type, Pass a numpy.ndarray'
+
+    N = len(data)
+    if method == 'EMD':
+        emd = EMD()
+        IMFs = emd.emd(normalize(data))
+    elif method == 'EEMD':
+        eemd = EEMD()
+        IMFs = eemd.eemd(normalize(data))
+    elif method == 'CEEMDAN':
+        ceemdan = CEEMDAN()
+        IMFs = ceemdan.ceemdan(normalize(data))
+
+    assert rescaling_imf > 0 and rescaling_imf <= len(IMFs), 'Invalid rescaling IMF'
+
+    output = {}
+    if test == 'apriori':
+        for idx,imf in enumerate(IMFs):
+            T = mean_period(imf)
+            energy_density = energy(imf)/N
+            sig_priori = significance_apriori(math.log(energy_density), T, N, alpha)
+            if sig_priori:
+                output[idx+1] = 1
+            else:
+                output[idx+1] = 0
+    else:
```

Comment:
I understand that we have checked earlier that `test in ('apriori', 'aposteriori')` so there are only two options. However, for readability and to not scroll code too much I'd suggest
```python
if test == 'apriori':
    ...
elif test == 'aposteriori':
    ...
else:
    raise AssertionError("Only 'apriori' and 'aposteriori' are allowed")
```
In such a case we can easily expand the test if needed.

Reply:
done